File: 02. Algorithms/01. Arrays/01. Reverse an Array/ReverseArray.java
// TODO: https://www.geeksforgeeks.org/write-a-program-to-reverse-an-array-or-string/

package array;

import java.util.Arrays;

public class ReverseArray {

	public static int[] reverseIteratively(int arr[], int length) {
		int s = 0, e = length - 1;
		while (s < e) {
			int temp = arr[s];
			arr[s] = arr[e];
			arr[e] = temp;
			s++;
			e--;
		}
		return arr;
	}

	public static void reverseRecursively(int arr[], int s, int e) {
		if (s >= e)
			return;
		int temp = arr[s];
		arr[s] = arr[e];
		arr[e] = temp;
		reverseRecursively(arr, ++s, --e);
	}

	public static void main(String[] args) {
		int arr[] = { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
		int len = arr.length;
		System.out.println("Original Array: " + Arrays.toString(arr));

//		reverseIteratively(arr, len);
		System.out.println("Iterative Reverse: " + Arrays.toString(arr));

		reverseRecursively(arr, 0, len - 1);
		System.out.println("Recursive Reverse: " + Arrays.toString(arr));
	}
}

File: 02. Algorithms/01. Arrays/01. Reverse an Array/Reverse_array.js
const myArr = [1,2,3,4,5,6];

function reverseArray(arr) {

  for(let i = 0; i < Math.floor(arr.length / 2); i++) {
    console.log(arr[i], arr[arr.length - 1 - i]);
    const temp = arr[i];
    arr[i] = arr[arr.length - 1 - i];
    arr[arr.length - 1 - i]  = temp;
    
  }

  return arr;
}
console.log(reverseArray(myArr));
